tasks/insights-client.yml

```diff
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Ensure required packages are installed
+  package:
+    name: insights-client
+    state: present
+  when: rhc_insights_state | d("present") == "present"
+
+- name: Call insights status
+  command: insights-client --status
+  ignore_errors: true
+  changed_when: false
+  register: __rhc_insights_status
+
+- name: Unregister insights-client
+  command: insights-client --unregister
+  when:
+    - rhc_insights.state | d("present") == "absent"
+    - ("unregistered" not in __rhc_insights_status.stdout)
+    - ("NOT" not in __rhc_insights_status.stdout)
```

Comment:
nitpicks: it is possible to remove the round brackets around the whole conditions

Reply:
well, you'll still need quoting - I guess you could try
```yaml
    - "unregistered" not in __rhc_insights_status.stdout
    - "NOT" not in __rhc_insights_status.stdout
```
and if that doesn't work, then use
```yaml
    - '"unregistered" not in __rhc_insights_status.stdout'
    - '"NOT" not in __rhc_insights_status.stdout'
```

Reply:
I opted for the second option, the first one didn't work